sdcm/utils/remotewebbrowser.py

```diff
@@ -0,0 +1,151 @@
+
+    @property
+    def browser(self):
+        return self._remote_browser
+
+    def _get_remote_browser(self):
+        """Get instance of remote browser
+
+        Get remote browser as instance of Selenium
+        remote webdriver running in containter described by RemoteWebDriverContainer
+
+        if self._use_tunnel true, start container with auto_ssh to up tunnel to
+        node were docker with remote selenium driver is running
+
+        :returns: Opened session to remote selenium browser
+        :rtype: {selenium.webdriver.Remote}
+        """
+        chrome_options = webdriver.ChromeOptions()
+
+        if not is_auto_ssh_running(self.remote_web_driver.name, self.remote_web_driver.node) and self._use_tunnel:
+            LOGGER.debug("Start autossh for Selenium remote webdriver ")
+            start_auto_ssh(self.remote_web_driver.name,
+                           self.remote_web_driver.node,
+                           self.remote_web_driver.remote_port,
+                           self.local_port,
+                           ssh_mode="-L")
+            LOGGER.debug("Waiting tunnel is up")
+            time.sleep(5)
+        if self._use_tunnel:
+            remote_browser_url = "http://127.0.0.1:{0.local_port}/wd/hub".format(self)
+        else:
+            remote_browser_url = "http://{0.external_address}:{0.remote_port}/wd/hub".format(self.remote_web_driver)
+
+        driver = webdriver.Remote(
+            command_executor=remote_browser_url,
+            options=chrome_options
+        )
+
+        return driver
+
+    def get_screenshot(self, url, screenshot_path, resolution="1920px*1280px"):
+        """get screenshot of url and save to file screenshot_path
+
+        Use remote selenium webdriver containter, open url
+        wait page loading for time and create screenshot
+
+        Arguments:
+            node {BaseNode} -- monitor node
+            url {str} -- url to load
+            screenshot_path {str} -- path to save screenshot in format 1920px*1280px
+
+        Keyword Arguments:
+            resolution {str} -- [description] (default: {"1920x1280"})
+        """
+
+        def wait_page_loading(_):
+            time.sleep(5)
+            if time.time() - start_time < self.load_page_timeout:
+                return False
+            else:
+                return True
+
+        resolution = resolution.split("*")
+        self._remote_browser.set_window_size(int(resolution[0][:-2]), int(resolution[1][:-2]))
+        LOGGER.info("Get screenshot of page %s", url)
+        self._remote_browser.get(url)
+        start_time = time.time()
+        try:
+            LOGGER.debug('Wait page loading ...')
+            WebDriverWait(self._remote_browser, self.load_page_timeout * 2).until(wait_page_loading)
+        finally:
+            LOGGER.debug('Create snapshot')
+            self._remote_browser.get_screenshot_as_file(screenshot_path)
+
+    def __del__(self):
```

Comment:
don't use __del__ for cleanups like that. you have little control on if it called at all.

I would do that explicitly in `get_screenshot` in a finally clause.

Reply:
removed.